extensions/pkg/controller/healthcheck/worker/nodes.go

```diff
@@ -159,10 +159,21 @@ func (h *DefaultHealthChecker) Check(ctx context.Context, request types.Namespac
 		for _, failedMachine := range deployment.Status.FailedMachines {
 			err := fmt.Errorf("machine %q failed: %s", failedMachine.Name, failedMachine.LastOperation.Description)
 			h.logger.Error(err, "Health check failed")
+
+			codes := []gardencorev1beta1.ErrorCode{}
+			unauthenticatedRegexp := regexp.MustCompile(`(?i)(InvalidAuthenticationTokenTenant|Authentication failed|AuthFailure|invalid character|invalid_client|query returned no results|InvalidAccessKeyId|cannot fetch token|InvalidSecretAccessKey|InvalidSubscriptionId)`)
+			unauthorizedRegexp := regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|SignatureDoesNotMatch|AuthorizationFailed|invalid_grant|Authorization Profile was not found|no active subscriptions|UnauthorizedOperation|not authorized|AccessDenied|OperationNotAllowed|Error 403|SERVICE_ACCOUNT_ACCESS_DENIED)`)
+			if unauthenticatedRegexp.MatchString(err.Error()) {
+				codes = append(codes, gardencorev1beta1.ErrorInfraUnauthenticated)
+			}
+			if unauthorizedRegexp.MatchString(err.Error()) {
+				codes = append(codes, gardencorev1beta1.ErrorInfraUnauthorized)
+			}
```

Comment:
Well, it's quite unfortunate that we still have provider-specific regexes here. We should get rid of them. Probably it makes more sense to let the extensions provide some regexes when instantiating the health check controller instead of hard-coding them here.

same below